common/lib/xmodule/xmodule/mongo_connection.py

```diff
@@ -0,0 +1,53 @@
+"""
+Common MongoDB connection functions.
+"""
+import pymongo
+from mongodb_proxy import MongoProxy
+
+
+# pylint: disable=bad-continuation
+def connect_to_mongodb(
+    db, host,
+    port=27017, tz_aware=True, user=None, password=None,
+    retry_wait_time=0.1, proxy=True, try_secondary_read=False, **kwargs
+):
+    """
+    Returns a MongoDB Database connection, optionally wrapped in a proxy. The proxy
+    handles AutoReconnect errors by retrying read operations, since these exceptions
+    typically indicate a temporary step-down condition for MongoDB.
+    """
+    # The MongoReplicaSetClient class is deprecated in Mongo 3.x, in favor of using
+    # the MongoClient class for all connections. Update/simplify this code when using
+    # PyMongo 3.x.
+    replicaset = kwargs.get('replicaSet', None)
+    if try_secondary_read and replicaset:
+        # Enable reading from secondary nodes in the MongoDB replicaset by using the
+        # MongoReplicaSetClient class.
+        # The 'replicaSet' parameter in kwargs is required for secondary reads.
+        # The read_preference should be set to a proper value, like SECONDARY_PREFERRED.
+        mongo_client_class = pymongo.MongoReplicaSetClient
+    else:
+        mongo_client_class = pymongo.MongoClient
```

Comment:
I added `try_secondary_read` merely to be able to override whether or not a secondary read was even attempted. I agree that it's not totally needed - and the secondary read (or not) could be driven by the configuration itself (via `replicaSet` and `read_preference`). It really just exists to prevent GridFS reads from using the secondary nodes - though that might just be fine as well.

On reflection, I'm going to remove this flag and let the doc_store_config drive the secondary reads.  @feanil @jibsheet Take note - you guys can see those settings and I can't.